mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,595 @@
+                Ok(h) => {
+                    let spending_tx = match Transaction::try_from(output_spend.spending_tx) {
+                        Ok(tx) => tx,
+                        Err(e) => return Err(FindWatchedOutputSpendError::TransactionConvertionErr(e.to_string())),
+                    };
+                    return Ok(Some((h, output_spend.input_index, spending_tx, height as u64)));
+                },
+                Err(e) => return Err(FindWatchedOutputSpendError::BlockHeaderDeserializeErr(e.to_string())),
+            }
+        }
+    }
+    Ok(None)
+}
+
+pub async fn get_best_header(best_header_listener: &ElectrumClient) -> EnableLightningResult<ElectrumBlockHeader> {
+    best_header_listener
+        .blockchain_headers_subscribe()
+        .compat()
+        .await
+        .map_to_mm(|e| EnableLightningError::RpcError(e.to_string()))
+}
+
+pub async fn update_best_block(
+    chain_monitor: Arc<ChainMonitor>,
+    channel_manager: Arc<ChannelManager>,
+    best_header: ElectrumBlockHeader,
+) {
+    {
+        let (new_best_header, new_best_height) = match best_header {
+            ElectrumBlockHeader::V12(h) => {
+                let nonce = match h.nonce {
+                    ElectrumNonce::Number(n) => n as u32,
+                    ElectrumNonce::Hash(_) => {
+                        return;
+                    },
+                };
+                let prev_blockhash = match sha256d::Hash::from_slice(&h.prev_block_hash.0) {
+                    Ok(h) => h,
+                    Err(e) => {
+                        log::error!("Error while parsing previous block hash for lightning node: {}", e);
+                        return;
+                    },
+                };
+                let merkle_root = match sha256d::Hash::from_slice(&h.merkle_root.0) {
+                    Ok(h) => h,
+                    Err(e) => {
+                        log::error!("Error while parsing merkle root for lightning node: {}", e);
+                        return;
+                    },
+                };
+                (
+                    BlockHeader {
+                        version: h.version as i32,
+                        prev_blockhash: BlockHash::from_hash(prev_blockhash),
+                        merkle_root: TxMerkleNode::from_hash(merkle_root),
+                        time: h.timestamp as u32,
+                        bits: h.bits as u32,
+                        nonce,
+                    },
+                    h.block_height as u32,
+                )
+            },
+            ElectrumBlockHeader::V14(h) => {
+                let block_header = match deserialize(&h.hex.into_vec()) {
+                    Ok(header) => header,
+                    Err(e) => {
+                        log::error!("Block header deserialization error: {}", e.to_string());
+                        return;
+                    },
+                };
+                (block_header, h.height as u32)
+            },
+        };
```

Comment:
I have something equivalent:
```rust
impl From<ElectrumBlockHeader> for SqlBlockHeader {
    fn from(header: ElectrumBlockHeader) -> Self {
        match header {
            ElectrumBlockHeader::V12(h) => {
                let block_hex = h.as_hex();
                let block_height = h.block_height.to_string();
                SqlBlockHeader {
                    block_height,
                    block_hex,
                }
            },
            ElectrumBlockHeader::V14(h) => {
                let block_hex = format!("{:02x}", h.hex);
                let block_height = h.height.to_string();
                SqlBlockHeader {
                    block_height,
                    block_hex,
                }
            },
        }
    }
}
```

Maybe we can have a common function and use it to avoid code repetition